src/mds/Server.cc

```diff
@@ -5722,6 +5722,32 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& value_str)
+{
+  if (value_str.length() < 1) {
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_value = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char last_char = tolower(value_str.back());
+  string value = value_str.substr(0, value_str.length() - 1);
+  if (size_value.find(last_char) != size_value.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(value)) * size_value[last_char];
+    } catch (boost::bad_lexical_cast const&) {
+      dout(10) << "invalid numerical value encountered: " << value << dendl;
```

Comment:
If the original `value_str` string is just **"g"**, then `last_char` will be `'g'` but `value` (coefficient) string will be the empty string.
So, the log message will look absurd after printing the empty string.

You might want to quote the `value` in the log message or specialize the log message for empty prefix (coefficient) string.